src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs

```diff
@@ -505,6 +525,64 @@ private static void WriteTransport(JsonWriter writer, string transportName, Tran
+            // structure from the low level properties
+            var existingRequestFeature = context.Features.Get<IHttpRequestFeature>();
+
+            var requestFeature = new HttpRequestFeature();
+            requestFeature.Protocol = existingRequestFeature.Protocol;
+            requestFeature.Method = existingRequestFeature.Method;
+            requestFeature.Scheme = existingRequestFeature.Scheme;
+            requestFeature.Path = existingRequestFeature.Path;
+            requestFeature.PathBase = existingRequestFeature.PathBase;
+            requestFeature.QueryString = existingRequestFeature.QueryString;
+            requestFeature.RawTarget = existingRequestFeature.RawTarget;
+            var requestHeaders = new Dictionary<string, StringValues>(existingRequestFeature.Headers.Count);
+            foreach (var header in existingRequestFeature.Headers)
+            {
+                requestHeaders[header.Key] = header.Value;
+            }
+            requestFeature.Headers = new HeaderDictionary(requestHeaders);
+
+            var existingConnectionFeature = context.Features.Get<IHttpConnectionFeature>();
+            var connectionFeature = new HttpConnectionFeature();
+
+            if (existingConnectionFeature != null)
+            {
+                connectionFeature.ConnectionId = existingConnectionFeature.ConnectionId;
+                connectionFeature.LocalIpAddress = existingConnectionFeature.LocalIpAddress;
+                connectionFeature.LocalPort = existingConnectionFeature.LocalPort;
+                connectionFeature.RemoteIpAddress = existingConnectionFeature.RemoteIpAddress;
+                connectionFeature.RemotePort = existingConnectionFeature.RemotePort;
+            }
+
+            // The response is a dud, you can't do anything with it anyways
+            var responseFeature = new HttpResponseFeature();
+
+            var features = new FeatureCollection();
+            features.Set<IHttpRequestFeature>(requestFeature);
+            features.Set<IHttpResponseFeature>(responseFeature);
+            features.Set<IHttpConnectionFeature>(connectionFeature);
+
+            // REVIEW: We could strategically look at adding other features but it might be better
+            // if we expose a callback that would allow the user to preserve HttpContext properties.
+
+            var newHttpContext = new DefaultHttpContext(features);
+            newHttpContext.TraceIdentifier = context.TraceIdentifier;
+            newHttpContext.User = context.User;
+
+            // Making request services function property could be tricky and expensive as it would require
+            // DI scope per connection.
+            newHttpContext.RequestServices = EmptyServiceProvider.Instance;
+
+            // REVIEW: This extends the lifetime of anything that got put into HttpContext.Items
+            newHttpContext.Items = context.Items.ToDictionary(p => p.Key, p => p.Value);
```

Comment:
Hmmm. To be honest I'm not 100% sure how `HttpContext` works in websockets land. Is there middleware that would place values in `Items`?

Reply:
Also, would this work to copy the dictionary:

`newHttpContext.Items = new Dictionary<object, object>(context.Items);`

Reply:
> Hmmm. To be honest I'm not 100% sure how HttpContext works in websockets land. Is there middleware that would place values in Items?

The HttpContext is fine in server sent events and websockets, it's in tact because the request doesn't complete until the connection is dead.

> Also, would this work to copy the dictionary:

I'll do that not sure why I LINQified this thing 😄